pkg/serviceaccounttoken/secret.go

```diff
@@ -153,14 +194,45 @@ func ServiceAccountSecret(ctx context.Context, sa *v1.ServiceAccount, secretList
 			logrus.Errorf("unable to delete secret [%s:%s]: %v", s.Namespace, s.Name, err)
 		}
 	}
+
 	return result, nil
 }
 
-func isSecretForServiceAccount(secret *v1.Secret, sa *v1.ServiceAccount) bool {
-	if secret.Type != v1.SecretTypeServiceAccountToken {
+func isSecretForServiceAccount(secret *corev1.Secret, sa *corev1.ServiceAccount) bool {
+	if secret.Type != corev1.SecretTypeServiceAccountToken {
 		return false
 	}
 	annotations := secret.Annotations
 	annotation := annotations[serviceAccountSecretAnnotation]
+
 	return sa.Name == annotation
 }
+
+func createServiceAccountSecret(ctx context.Context, sa *corev1.ServiceAccount, secretLister secretLister, secretClient clientv1.SecretInterface, lockKey string) (*corev1.Secret, error) {
+	if lockKey != "" {
```

Comment:
AFAIU this never happens, IIUC, lockKey is always (expected to be) non-empty.

I'd suggest to remove the condition, as otherwise the lock would seem bypassable when it is not designed to ever be bypassed